bot/extensions/clashofcode/views.py

```diff
@@ -0,0 +1,178 @@
+
+    @discord.ui.button(
+        label="Create Clash of Code", style=discord.ButtonStyle.green, emoji="📝", custom_id=CREATE_CUSTOM_ID, row=3
+    )
+    async def create_coc(self, interaction: core.InteractionType, _button: ui.Button):
+        selected_languages = discord.utils.get(self.children, custom_id=self.LANGUAGES_CUSTOM_ID).values
+        selected_modes = discord.utils.get(self.children, custom_id=self.MODES_CUSTOM_ID).values
+
+        if len(selected_languages) == 25:
+            selected_languages = []
+        if not selected_modes:
+            selected_modes = modes
+
+        await interaction.response.defer()
+        await interaction.delete_original_response()
+
+        data = await coc_client.request(
+            "ClashOfCode",
+            "createPrivateClash",
+            [coc_client.codingamer.id, selected_languages, selected_modes],
+        )
+        handle = data["publicHandle"]
+
+        # Check if it has been too long since the last clash
+        if coc_helper.session and coc_helper.last_clash + 1800 < int(time.time()):
+            coc_helper.session = False
+
+        await interaction.channel.send(
+            (
+                f"**Hey, {interaction.user.mention} is hosting a Clash Of Code game!**\n"
+                f"Mode{'s' if len(selected_modes) > 1 else ''}: {', '.join(selected_modes).title()}\n"
+                f"Programming languages: {', '.join(selected_languages) if selected_languages else 'All'}\n"
+                f"Join here: <https://www.codingame.com/clashofcode/clash/{handle}>\n"
+                f"{f'<@&{settings.coc.role_id}>' if not coc_helper.session else ''}"
+            ),
+            allowed_mentions=discord.AllowedMentions(users=True, roles=True),
+        )
+
+        coc_helper.session = True
+        coc_helper.last_clash = time.time()
+
+        clash = await coc_client.get_clash_of_code(handle)
+        coc_helper.clash = clash
+        while not clash.started:
+            await asyncio.sleep(10)  # wait 10s to avoid flooding the API
+            clash = await coc_client.get_clash_of_code(handle)
+
+        players = [player.pseudo for player in clash.players]
+        start_message = await interaction.channel.send(embed=em(clash.mode, ", ".join(players)))
+
+        test_session_handle = None
+        try:
+            # Get test session handle
+            data = await coc_client.request("ClashOfCode", "startClashTestSession", [coc_client.codingamer.id, handle])
+            test_session_handle = data["handle"]
+        except HTTPError as e:
+            log.error(f"Failed to start test session for {handle}:", exc_info=e)
+
+        if test_session_handle:
+            if selected_languages:
+                language = selected_languages[0]
+            else:
+                language = "Python3"
+
+            try:
+                # Submit an empty solution to not hold up the clash once everyone has submitted
+                await coc_client.request(
```

Comment:
Maybe (if there's any reason to) we should wait with submitting ours until the game is over?